### PR TITLE
docs(quickstart): update, add common errors and prereq installation

### PR DESCRIPTION
**Pull request by qbzzt**
1. Unify with the TypeScript templates quickstart (same content)
2. Add common errors

### CODE REVIEW

File: docs/pages/quickstart.mdx [hunk 1]
```diff
@@ -1,20 +1,245 @@
...
+In most cases the easiest way to get started is to use
+[the TypeScript templates](https://github.com/latticexyz/mud/tree/main/templates) provided by [Lattice](https://lattice.xyz/).
+
+## Prerequisites
+
+To install the TypeScript templates you need:
+
+- [Node.js v18](https://nodejs.org/en/download/package-manager).
+  Note that version 18 is _required_.
+  We do not support older or newer versions.
```
Comment (alvrs) [inline docs/pages/quickstart.mdx:15]:
```suggestion
  We do not support older or newer versions at the moment.
```

File: docs/pages/templates/typescript/_meta.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 export default {
-  "getting-started": "Getting Started",
```
Comment (holic) [inline docs/pages/templates/typescript/_meta.js:2]: reminder that we should always add a redirect if we break a link

**Review by alvrs — APPROVED** [review]

File: docs/pages/templates/typescript/_meta.js (hunk 2 above)
Comment (qbzzt) [reply docs/pages/templates/typescript/_meta.js:2]: Thank you for reminding me. Done.

**Review by alvrs — DISMISSED** [review]

File: docs/next.config.mjs [hunk 3]
```diff
@@ -221,6 +221,11 @@ export default withNextra({
         destination: "/guides/extending-a-world",
         permanent: false,
       },
+      {
+        source: "/templates/typescript/getting-started",
+        destination: "/quickstart",
+        permanent: true,
```
Comment (holic) [inline docs/next.config.mjs:227]: always false for these!
```suggestion
        permanent: false,
```

**Review by alvrs — APPROVED** [review]

**Pull request merged**